src/environment/ale_state.hpp

```diff
@@ -21,9 +21,10 @@
 #include "../emucore/OSystem.hxx"
 #include "../emucore/Event.hxx"
 #include <string>
-#include "../games/RomSettings.hpp"
 #include "../common/Log.hpp"
 
+class RomSettings;
```

Comment:
Where is this needed? It's bad to use a class forward declaration, but worse to put it in the header file.

Reply:
Well, basically, I can't manage to get rid of it. It is being used, in `stella_environment.hpp`, on the functions `load`,  `save`, and in the constructor. I couldn't find a way to get rid of this easily. Am I missing something? There's a long time since I did anything in C++.

Reply:
This used to #include RomSettings.hpp, which would avoid the forward declaration. So you must have done this to break the circular dependency RomSettings -> StellaEnvironment -> RomSettings. Hopefully if we break that we can revert to `#include "../games/RomSettings.hpp"`